### PR TITLE
Finish user signup

### PR DESCRIPTION
**Pull request by nguyenquocthong**

### CODE REVIEW

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -1,6 +1,14 @@
 @import "bootstrap-sprockets";
 @import "bootstrap";
+/* mixins, variables, etc. */
 
+$gray-medium-light: #eaeaea;
+
+@mixin box_sizing {
+  -moz-box-sizing:    border-box;
```
Comment (dat-hedspi) [inline app/assets/stylesheets/custom.scss]: ident

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,4 +1,25 @@
 class UsersController < ApplicationController
+  def show
+    @user = User.find params[:id]
+  end
+
   def new
+  	@user = User.new
```
Comment (dat-hedspi) [inline app/controllers/users_controller.rb]: ident

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,4 +1,25 @@
...
+    else
+      render "new"
+    end
   end
+
+  private
+
+    def user_params
+      params.require(:user).permit :name, :email, :password,
+                                   :password_confirmation
```
Comment (dat-hedspi) [inline app/controllers/users_controller.rb]: ident

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -10,8 +10,12 @@
   <body>
     <%= render "layouts/header" %>
     <div class="container">
+      <% flash.each do |message_type, message| %>
+        <div class="alert alert-<%= message_type %>"><%= message %></div>
+      <% end %>
       <%= yield %>
     <%= render "layouts/footer" %>
+    <%= debug params if Rails.env.development? %>
```
Comment (dat-hedspi) [inline app/views/layouts/application.html.erb]: ident 2 dòng trên

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,4 +1,24 @@
...
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
+      redirect_to @user
+    else
+      render "new"
+    end
   end
+
+  private
+    def user_params
```
Comment (vietmt) [inline app/controllers/users_controller.rb:20]: private
def user_params

File: app/views/shared/_error_messages.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize @user.errors.count, "error" %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (vietmt) [inline app/views/shared/_error_messages.html.erb:7]: indent

**Pull request merged**